src/main/java/com/google/devtools/build/lib/bazel/BazelRepositoryModule.java

```diff
@@ -462,6 +462,16 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
       }
       httpDownloader.setMaxAttempts(repoOptions.httpConnectorAttempts);
       httpDownloader.setMaxRetryTimeout(repoOptions.httpConnectorRetryMaxTimeout);
+      if (repoOptions.httpMaxParallelDownloads > 0) {
+        httpDownloader.setMaxParallelDownloads(repoOptions.httpMaxParallelDownloads);
+      } else {
+        env.getReporter()
```

Comment:
Failures are usually preferred over warnings, both to reduce the amount of "warning spam" and also because it makes it easier to change the behavior later in a compatible way (for example, we could introduce new logic that chooses a reasonable default based on host machine info if this is set to 0).

See https://github.com/bazelbuild/bazel/blob/0e220f03f154cbb5e52190e1085b01173f301c02/src/main/java/com/google/devtools/build/lib/bazel/BazelRepositoryModule.java#L396-L404 for an example.